File: v3-sandbox/prototype-keys-java/src/main/java/org/hiero/keys/io/KeyContainer.java
package org.hiero.keys.io;

import org.hiero.keys.KeyType;

/**
 * Supported container formats.
 */
public enum KeyContainer {
    PKCS8, // private keys
    SPKI;  // public keys

    public boolean supportsType(final KeyType type) {
        return switch (this) {
            case PKCS8 -> type == KeyType.PRIVATE;
            case SPKI -> type == KeyType.PUBLIC;
        };
    }
}

File: v3-sandbox/prototype-python/src/hiero_sdk_v3/keys/io/key_container.py
from enum import Enum
from ..key_type import KeyType


class KeyContainer(Enum):
    PKCS8 = "PKCS8"
    SPKI = "SPKI"

    def supports_type(self, key_type: KeyType) -> bool:
        return key_type == KeyType.PRIVATE if self == KeyContainer.PKCS8 else key_type == KeyType.PUBLIC
